nntrainer/utils/node_exporter.h

```diff
@@ -147,5 +114,90 @@ class Exporter {
   bool is_exported; /**< boolean to check if exported */
 };
 
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam Callable generic lambda to be called during iteration
+ * @tparam Ts types from tuple
+ * @param c callable gerneric labmda
+ * @param tup tuple to be iterated
+ * @return void
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<I == sizeof...(Ts), void>::type
+iterate_prop(Callable &&c, const std::tuple<Ts...> &tup) {
+  // end of recursion;
+}
+
+/**
+ * @brief base case of iterate_prop, iterate_prop iterates the given tuple
+ *
+ * @tparam I size of tuple(automated)
+ * @tparam Callable generic lambda to be called during iteration
+ * @tparam Ts types from tuple
+ * @param c callable gerneric labmda
+ * @param tup tuple to be iterated
+ * @return not used
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type
+iterate_prop(Callable &&c, const std::tuple<Ts...> &tup) {
+  c(std::get<I>(tup), I);
+
+  iterate_prop<I + 1>(c, tup);
+}
+
+/**
+ * @copydoc  template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type iterate_prop(Callable
+&&c, const std::tuple<Ts...> &tup)
+ */
+template <size_t I = 0, typename Callable, typename... Ts>
+typename std::enable_if<(I < sizeof...(Ts)), void>::type
+iterate_prop(Callable &&c, std::tuple<Ts...> &tup) {
+  c(std::get<I>(tup), I);
+
+  iterate_prop<I + 1>(c, tup);
+}
+
+/**
+ * @brief load property from the api formatted string ({"key=value",
+ * "key1=value1"})
+ *
+ * @tparam Ts prop type
+ * @param string_vector api formatted string;
+ * @param[out] props props to be iterated
+ * @return std::vector<std::string> vector of string that is not used while
+ * setting the property
+ */
+template <typename... Ts>
+std::vector<std::string>
+load_properties(const std::vector<std::string> &string_vector,
+                std::tuple<Ts...> &props) {
+
+  std::vector<std::string> left = string_vector;
+
+  auto callable = [&left](auto &&prop, size_t index) {
+    std::string prop_key = std::remove_reference_t<decltype(prop)>::key;
+
+    for (auto iter = left.begin(); iter < left.end(); ++iter) {
+      std::string key, value;
+      int status = getKeyValue(*iter, key, value);
```

Comment:
in this logic, for every prop_key, the string is being parse into key and value in each loop.
Inverse the order of loops to parse the string just once, and re-use key-value pair.

Reply:
Done

Reply:
Updated to do less operation.